.github/workflows/awesome_workflow.yml

```diff
@@ -15,14 +12,14 @@ jobs:
       - uses: actions/setup-python@v2
       - name: requirements
         run: | 
-          sudo apt -qq -y update
-          sudo apt -qq install clang-tidy-10 clang-format-10
+          sudo apt-get -qq update
+          sudo apt-get -qq install clang-tidy clang-format
         # checks are passing with less errors when used with this version. 
         # The default installs v6.0 which did not work out well in my tests
       - name: Setup Git Specs
         run: |
-          git config --global user.name David Leal
-          git config --global user.email 'Panquesito7@users.noreply.github.com'
+          git config --global user.name github-actions[bot]
+          git config --global user.email 'github-actions@users.noreply.github.com'
           git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/$GITHUB_REPOSITORY
```

Comment:
Also this line is not needed now I think - you can just use the default `git push` and it should work.
```suggestion
```

This might also be the reason for all then issues of permissions throughout the workflow - remove these instances (line 58) and it should be fine

Reply:
Also, the line 56

```sh
git diff --diff-filter=dr --name-only origin/master > git_diff.txt
```

is quite old way of referencing origin branches and potentially problematic. You should change the `checkout` action to `v3` since then use
```sh
git diff --diff-filter=dr --name-only origin master > git_diff.txt
```

the `/` character in the middle of `origin` and `master` is not correct in newer versions of `git`